wandb/sdk/data_types/base_types/wb_value.py

```diff
@@ -25,15 +24,8 @@ def _server_accepts_client_ids() -> bool:
     # AS OF NOW, 2024/11/06, we assume that all customer's server deployments accept
     # client IDs.
 
-    if util._is_offline():
-        # If there are any users with issues on an older backend, customers can disable the
-        # setting `allow_offline_artifacts` to revert the SDK's behavior back to not
-        # using client IDs in offline mode.
-        if wandb.run and not wandb.run.settings.allow_offline_artifacts:
-            return False
-        # Assume client IDs are accepted
-        else:
-            return True
+    if run and run.offline:
```

Comment:
what happens if run is None?

Reply:
In that case, we treat it as online. The only place it can be None is when using `Artifact.add()`.

I'm actually not sure if this is what we want... maybe if run is None, we should assume client IDs are accepted and return True instead? But that bypasses the `allow_offline_artifacts` setting, making it useless.

This feels like a design issue with artifacts. We shouldn't need to know whether the server accepts client IDs until we try to sync.

Reply:
Agreed re-design!